src/get_family_names.py

```diff
@@ -70,13 +72,23 @@ def flatten_parents(df):
 
 
 def get_all_congress_people_ids():
+    print('Fetching all congresspeople ids', end='\r')
     ids_series = [read_csv(name)['congressperson_id']
                   for name in ['current-year', 'last-year', 'previous-years']]
     return list(pd.concat(ids_series).unique())
 
 
+def find_latest_date():
+    date_regex = re.compile('\d{4}-\d{2}-\d{2}')
+
+    matches = (date_regex.findall(f) for f in os.listdir(DATA_DIR))
+    dates = (l[0] for l in matches if l)
```

Comment:
Not sure if this is the best approach because we could have other documents newer than the trio `current-year`/`last-year`/`previous-years` and that would fool us into getting the wrong date to that trio. For example, in a directory like that:

```console
2016-10-12-congressperson_relatives.xz
2016-08-08-current-year.xz
...
```

`find_latest_date()` would return `2016-10-12`, but we won't be able to find `2016-10-12-current-year.xz`.

In [another script](https://github.com/datasciencebr/serenata-de-amor/blob/master/src/fetch_receipts.py#L20) I used this regular expression if it helps: `r'^[\d-]{11}(current-year|last-year|previous-years).xz$'` (11 number or dash characters, then one of the file names we're looking for). Not the most precise regular expression ever, but does the job…

Maybe grouping the date and fetching only the required group will be even better. For example:

```python
regex = r'^(?P<date>\d{4}-\d{2}-\d{2})(-)(current-year|last-year|previous-years).xz$'`
results = re.match(regex, '2016-08-08-current-year.xz')
results.group('date')  # '2016-08-08'
```

Anyway, just brainstorming here… ; )